src/System.Net.NameResolution/src/System/Net/DnsResolveAsyncResult.cs

```diff
@@ -0,0 +1,25 @@
+namespace System.Net
+{
+    internal class DnsResolveAsyncResult : ContextAwareResult
+    {
+        // Forward lookup
+        internal DnsResolveAsyncResult(string hostName, object myObject, bool includeIPv6, object myState, AsyncCallback myCallBack) :
+            base(myObject, myState, myCallBack)
+        {
+            this.hostName = hostName;
+            this.includeIPv6 = includeIPv6;
+        }
+
+        // Reverse lookup
+        internal DnsResolveAsyncResult(IPAddress address, object myObject, bool includeIPv6, object myState, AsyncCallback myCallBack) :
+            base(myObject, myState, myCallBack)
+        {
+            this.includeIPv6 = includeIPv6;
+            this.address = address;
+        }
+
+        internal readonly string hostName;
+        internal bool includeIPv6;
+        internal IPAddress address;
```

Comment:
Nit: these should be prefixed with _

Reply:
Can these be made readonly?

Reply:
This was existing code moved from DNS.cs, but I will make the changes